lib/Backend/LowerMDShared.cpp

```diff
@@ -9153,32 +9153,27 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
             }
 
             // ROUNDSD srcCopy, srcCopy, round_mode
-            IR::Opnd * roundMode;
-#ifdef ENABLE_WASM
-            if (instr->m_opcode == Js::OpCode::Trunc_A)
+            IR::Opnd * roundMode = nullptr;
+
+            switch (instr->m_opcode)
             {
+#ifdef ENABLE_WASM
+            case Js::OpCode::Trunc_A:
                 roundMode = IR::IntConstOpnd::New(0x03, TyInt32, this->m_func);
-            }
-            else if (instr->m_opcode == Js::OpCode::Nearest_A)
-            {
+                break;
+            case Js::OpCode::Nearest_A:
                 roundMode = IR::IntConstOpnd::New(0x00, TyInt32, this->m_func);
-            }
-            else
+                break;
 #endif //ENABLE_WASM
-            {
-                if (isNotCeil)
-                {
-                    roundMode = IR::IntConstOpnd::New(0x01, TyInt32, this->m_func);
-                }
-                else if (instr->GetDst()->IsInt32() || instr->m_opcode != Js::OpCode::InlineMathFloor)
-                {
-                    roundMode = IR::IntConstOpnd::New(0x02, TyInt32, this->m_func);
-                }
-                else
-                {
-                    roundMode = IR::IntConstOpnd::New(0x03, TyInt32, this->m_func);
```

Comment:
As far as I can tell. InlineMathRound and InlineMathFloor (i.e., != InlineMathCeil) are both handled by the first if-clause. Thus, the only remaining case is InlineMathCeil for which != InlineMathFloor always evaluates to true.